recipes/frugen/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import get, rmdir, apply_conandata_patches, export_conandata_patches
+import os
+
+class FrugenConan(ConanFile):
+    name = "frugen"
+    description = "IPMI FRU Information generator / editor tool and library"
+    license = ("Apache-2.0", "GPL-2.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://codeberg.org/IPMITool/frugen"
+    topics = ("hardware", "ipmi", "fru")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_json": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_json": True
+    }
+    implements = ["auto_shared_fpic"]
+
+    exports_sources = "CMakeLists.txt", "*.c", "*.h"
+
+    def validate(self):
+        if self.settings.os in ["Windows", "Macos"]:
+            raise ConanInvalidConfiguration("Unsupported os. This package currently does not support Macos and Windows")
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_json:
+            self.requires("json-c/0.18")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIB"] = self.options.shared
+        tc.cache_variables["ENABLE_JSON"] = self.options.with_json
+        tc.cache_variables["BINARY_STATIC"] = not self.options.shared
+        tc.cache_variables["BINARY_32BIT"] = False
+        tc.cache_variables["JSON_STATIC"] = False
```

Comment:
This is a bit confusing, I'd remove this line, as the patch is basically overriding the upstream behaviour and making this flag useless
```suggestion
```

Reply:
good point, will fix